### ISSUE TITLE
Pod rejection errors and unexpected MIG slice counts

### ISSUE THREAD

**Issue by tardieu** [openshift/instaslice-operator#96]
When following the Kueue+InstaSlice demo #92, I expect 7 pods to run to completion with no more than 3 pods running at a time to due the configuration of the cluster queue. While I observed no violation of the max concurrency, in many instances I observed a couple of unexpected outcomes:
1. One or several pods report the status `OutOfnvidia.com/mig-1g.5gb` with the message `Pod was rejected: Node didn''t have enough resource: nvidia.com/mig-1g.5gb, requested: 1, used: 0, capacity: 0`
2. The count of `nvidia.com/mig-1g.5gb` resources in the node capacity no longer matches the count of `org.instaslice/*` resources in the node capacity. More specifically the former is lower than the latter and lower than the number of slices required to run pending ungated pods.

This is running Kind 1.31 on rancher desktop 1.15.1 on Sonoma. The demo scenario relies on fake GPUs and InstaSlice [emulator](https://github.com/openshift/instaslice-operator/blob/main/docs/emulator.md) mode.

**Comment by tardieu** [comment]
I encounter the same node capacity mismatch issue ( nvidia.com/mig-1g.5gb count vs org.instaslice/* ) on kind 1.30.4:
```sh
kind create cluster --image kindest/node:v1.30.4@sha256:976ea815844d5fa93be213437e3ff5754cd599b040946b5cca43ca45c2047114
```

**Comment by tardieu** [comment]
My suspicion at this point is that in this scenario InstaSlice may destroy the slice intended for a pod right after ungating the pod leading to the 3 scenarios I have observed:
1. The pod was scheduled and started by the kubelet before the deletion. It appears to run successfully but only because the untimely deletion of the slice does not affect the sleep command running in the pod.
2. The pod remains pending because the slice has been deleted before the scheduler had a chance to schedule the pod.
3. The pod was scheduled but the kubelet reports an error because the slice was gone in-between the scheduler scheduling and the kubelet doing its thing.

In these emulated scenarios, the creation/deletion of the slice boils down to the addition and removal of one unit of the MIG resource in the node capacity.

**Comment by asm582** [comment]
Thanks for this issue. scenario 2 can be reproduced and we have a PR for it #99. Scenario 3 could result from dangling org.instaslice resource on the node which is causes emulator to perform excessive deletes

**Comment by asm582** [comment]
Solved by #121

**Comment by asm582** [comment]
With new design changes, I am not sure if this issue is valid
